### PR TITLE
CB-9393: Documenting how to embed a SystemWebView 

### PR DESCRIPTION
**Pull request by infil00p**
This is designed with Android Studio in mind.

### CODE REVIEW

**Comment by riknoll** [comment]
Awesome! I'll give this a try when I get a moment.

File: www/docs/en/dev/guide/platforms/android/webview.md [hunk 1]
```diff
@@ -29,106 +29,73 @@ components can communicate with each other, see Application Plugins.
...
 SDK installed before you attempt the more unusual development option
-of embedding a WebView.  Starting with Cordova 1.9, the Android
-platform relies on a `CordovaWebView` component, which builds on a
-legacy `CordovaActivity` component that pre-dates the 1.9 release.
+of embedding a WebView.  Starting with Cordova 4.0, the Android
+platform relies on a `SystemWebView` component.
 
 1. To follow these instructions, make sure you have the latest Cordova
    distribution. Download it from
    [cordova.apache.org](http://cordova.apache.org) and unzip its
```
Comment (jcesarmobile) [inline www/docs/en/dev/guide/platforms/android/webview.md]: It's hard to find where to download it. Maybe the link should point to dist page https://dist.apache.org/repos/dist/release/cordova/platforms/ or github releases page https://github.com/apache/cordova-android/releases

File: www/docs/en/dev/guide/platforms/android/webview.md [hunk 2]
```diff
@@ -29,106 +29,73 @@ components can communicate with each other, see Application Plugins.
...
-1. Navigate to the Android package's `/framework` directory and run
-   `ant jar`. It creates the Cordova `.jar` file, formed as
-   `/framework/cordova-x.x.x.jar`.
+1. Create a project in Android Studio
 
-1. Copy the `.jar` file into the Android project's `/libs` directory.
+1. Copy the framework directory into your project's root directory and name it
+CordovaLib
+
+1. Add that project as a library project to your main project.
```
Comment (jcesarmobile) [inline www/docs/en/dev/guide/platforms/android/webview.md]: maybe explain how to do it. I'm familiar with Android, but not so much with Android Studio and I didn't know how to do it.

File: www/docs/en/dev/guide/platforms/android/webview.md [hunk 3]
```diff
@@ -29,106 +29,73 @@ components can communicate with each other, see Application Plugins.
...
 
 1. Add the following to the application's `/res/xml/main.xml` file,
    with the `layout_height`, `layout_width` and `id` modified to suit
    the application:
 
-        <org.apache.cordova.CordovaWebView
+        <org.apache.cordova.SystemWebView
```
Comment (jcesarmobile) [inline www/docs/en/dev/guide/platforms/android/webview.md]: It's org.apache.cordova.engine.SystemWebView, not org.apache.cordova.SystemWebView

File: www/docs/en/dev/guide/platforms/android/webview.md [hunk 4]
```diff
@@ -29,106 +29,73 @@ components can communicate with each other, see Application Plugins.
...
-   following:
+        private CordovaInterfaceImpl iface = new CordovaInterfaceImpl(this);
+
+1. Add the following to intialize the WebView
+
+        //Set up the webview
+        ConfigXmlParser parser = new ConfigXmlParser();
+        parser.parse(this);
+
+        SystemWebView webView = (SystemWebView) findViewById(R.id.WebViewComponent);
```
Comment (jcesarmobile) [inline www/docs/en/dev/guide/platforms/android/webview.md]: you used the tutorialView id, not WebViewComponent

File: www/docs/en/dev/guide/platforms/android/webview.md [hunk 5]
```diff
@@ -29,106 +29,73 @@ components can communicate with each other, see Application Plugins.
...
+        webInterface = new CordovaWebViewImpl(new SystemWebViewEngine(webView));
```
Comment (jcesarmobile) [inline www/docs/en/dev/guide/platforms/android/webview.md]: Youd didn't create the webInterface variable before.

File: www/docs/en/dev/guide/platforms/android/webview.md [hunk 6]
```diff
@@ -29,106 +29,73 @@ components can communicate with each other, see Application Plugins.
...
+        webInterface.init(iface, parser.getPluginEntries(), parser.getPreferences());
+
+        loadUrl(parser.getLaunchUrl());
```
Comment (jcesarmobile) [inline www/docs/en/dev/guide/platforms/android/webview.md]: I think it should be webView.loadUrl(parser.getLaunchUrl());

File: www/docs/en/dev/guide/platforms/android/webview.md [hunk 7]
```diff
@@ -29,106 +29,73 @@ components can communicate with each other, see Application Plugins.
...
+
+
+1. Copy the application's HTML and JavaScript files to the Android
```
Comment (jcesarmobile) [inline www/docs/en/dev/guide/platforms/android/webview.md]: where do we get the cordova.js? There is one on /bin/templates/project/assets/

File: www/docs/en/dev/guide/platforms/android/webview.md [hunk 8]
```diff
@@ -29,106 +29,73 @@ components can communicate with each other, see Application Plugins.
...
+   project's `app/src/main/assets/www` directory.
+
+1. Copy the `config.xml` file from `/framework/res/xml` to the
```
Comment (jcesarmobile) [inline www/docs/en/dev/guide/platforms/android/webview.md]: There is no config.xml on /framework/res/xml. I've found two, one on test/res/xml and another on /bin/templates/project/res/xml, I suppose the good one is the one on bin, right?

**Comment by jcesarmobile** [comment]
I think this should be added too:

`@Override
    public void onDestroy() {
        if (this.webInterface!=null) {
            this.webInterface.handleDestroy();
        }
        super.onDestroy();
    }`

If I don't add it, when I use navigator.app.exitApp(); I get an ugly message like this:
"MainActivity has leaked IntentReceiver org.apache.cordova.engine.SystemWebViewEngine$2@e41c27f that was originally registered here. Are you missing a call to unregisterReceiver()?..."

**Comment by infil00p** [comment]
Redoing this pull request on a whole new branch, because this one-off is  not going to work with all the changes @jcesarmobile is suggesting.

**Pull request closed without merging**